conn.go

```diff
@@ -723,22 +723,22 @@ func (c *deadlineWriter) Write(p []byte) (int, error) {
 	return c.w.Write(p)
 }
 
-func newWriteCoalescer(conn net.Conn, timeout time.Duration, d time.Duration, quit <-chan struct{}) *writeCoalescer {
+func newWriteCoalescer(ctx context.Context, conn net.Conn, timeout time.Duration, d time.Duration) *writeCoalescer {
```

Comment:
What about deriving a context here, storing  cancel function and call that cancel stop instead of closing w.quit? This would limit that knowledge of  how to stop things to newWriteCoalescer and stop and keep writeFlusher the same.

That would also get rid of the danger of messing up channel sends, since the context package does that well enough already.